src/current-user.js

```diff
@@ -632,6 +636,45 @@ export class CurrentUser {
     return userPresenceSub.connect()
   }
 
+  enablePushNotifications(config = {}) {
+    try {
+      return this.beamsInstanceInitFn({
+        serviceWorkerURL: config.serviceWorkerURL || DEFAULT_SERVICE_WORKER_URL,
+      })
+        .then(beamsClient => {
+          beamsClient.start()
+          return beamsClient
+        })
+        .then(beamsClient => {
+          const fetchBeamsToken = userId =>
```

Comment:
This could handle/report errors better (bad status code, bad JSON etc.)

Reply:
Yes. I did a very basic logging and context-augmentation in 01a586b.

I reckon this is all the customer needs to know -- we are the ones responsible for this endpoint to work correctly, and have the correct integration (so handling status code differently seems be too much?). What do you think?

Reply:
If we get a bad status code the response will be a Chatkit Error JSON object.
Won't this just get passed to the Beams SDK rather than the token object?
Would lead to some unhelpful error messages. Maybe better to throw an error with a more helpful message that could be sent to support?